src/main/java/TicTacToe.java

```diff
@@ -0,0 +1,327 @@
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+// Strategy pattern.
+// The computer player's behavior/strategy can be replaced by inheriting from the interface below
+// Also, the human player's behavior inherits from the same interface
+// This also makes it easy to modify the game for 2 human players, 2 computer players etc.
+
+interface MoveMethod
+{
+    public int move();

```

Comment:
Possibly rename this to MoveStrategy as method is an overloaded word.

The interface would be better if we return a Move object versus just an int. That way its really clear what it is doing and that object can be passed around.